test/unit/lib/plugins/aws/provider.test.js

```diff
@@ -1098,6 +1098,7 @@ aws_secret_access_key = CUSTOMSECRET
       });
       const modulesCacheStub = {
         'child-process-ext/spawn': spawnExtStub,
+        './lib/utils/telemetry/generatePayload.js': async () => ({}),
```

Comment:
Why do we need to do that?

Reply:
The reason for that is the fact that `generatePayload` internally calls `isNpmGlobal` util which uses `spawn` call. For all these `docker`-based tests, we're mocking all `spawn` calls as well, which was broken due to an extra unexpected call from `generatePayload`. Previously it wasn't visible, because `generatePayload` was called only for `deploy` and these tests use `package` command. This changed in this PR. I was also considering another approach, e.g. disabling generation altogether when analytics are disabled. What do you think?

Reply:
I see, `runServelress has also `shouldStubSpawn` option, through which automatically all `spawn` commands are stubbed to be noop. Maybe using just that will do the job? (then no `modulesCacheStub` option is needed)

Reply:
I know that functionality, but some tests are assessed based on specific behavior of spawn (error scenarios), which means that functionality unfortunately doesn't solve the problem there.

Reply:
Ok, if that's the case, then all is good.